images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -47,14 +50,20 @@ function Set-JavaPath {
     }
 }
 
-function Install-JavaFromAdoptOpenJDK {
+function Install-JavaJDK {
     param(
         [string] $JDKVersion,
-        [string] $Architecture = "x64"
+        [string] $Architecture = "x64",
+        [string] $FullVendorName
     )
 
-    # Get Java version from adopt openjdk api
-    $assetUrl = Invoke-RestMethod -Uri "https://api.adoptopenjdk.net/v3/assets/latest/${JDKVersion}/hotspot"
+    # Get Java version from api
+    if ($FullVendorName -eq "Termurin-Hotspot") {
+        $assetUrl = Invoke-RestMethod -Uri "https://api.adoptium.net/v3/assets/latest/${JDKVersion}/hotspot"
+    }
+    else {
```

Comment:
Could you please convert this branch to the elseif to check exact Adoptium name and throw exception if unknown vendor is passed?
I would prefer to fail the build if vendor is invalid instead of silently installation of Adopt

```
if ($FullVendorName -eq "Temurin-Hotspot") {

} elseif ($FullVendorName -eq "Adopt") {

} else {
  throw "Invalid vendor name. 'Adopt' and 'Temurin-Hotspot' are allowed values"
}
```

Alternative will be make powershell validation of function params: https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/validateset-attribute-declaration?view=powershell-7.1#parameters

Reply:
Done